server/api/support.py

```diff
@@ -0,0 +1,174 @@
+import uuid
+import secrets
+from urllib.parse import urlparse
+from flask import jsonify, request, session, redirect
+from auth0.v3.authentication import GetToken
+from auth0.v3.management import Auth0
+from auth0.v3.exceptions import Auth0Error
+from werkzeug.exceptions import Conflict
+
+from . import api
+from ..models import *  # pylint: disable=wildcard-import
+from ..database import db_session
+from ..auth import (
+    restrict_access_superadmin,
+    set_loggedin_user,
+    UserType,
+)
+from ..config import (
+    AUDITADMIN_AUTH0_BASE_URL,
+    AUDITADMIN_AUTH0_CLIENT_ID,
+    AUDITADMIN_AUTH0_CLIENT_SECRET,
+)
+from ..util.jsonschema import validate
+
+AUTH0_DOMAIN = urlparse(AUDITADMIN_AUTH0_BASE_URL).hostname
+
+
+def auth0_get_token() -> str:
+    response = GetToken(AUTH0_DOMAIN).client_credentials(
+        AUDITADMIN_AUTH0_CLIENT_ID,
+        AUDITADMIN_AUTH0_CLIENT_SECRET,
+        f"https://{AUTH0_DOMAIN}/api/v2/",
+    )
+    return str(response["access_token"])
+
+
+def auth0_create_audit_admin(email: str) -> str:
+    token = auth0_get_token()
+    auth0 = Auth0(AUTH0_DOMAIN, token)
+    try:
+        user = auth0.users.create(
+            dict(
+                email=email,
+                password=secrets.token_urlsafe(),
+                connection="Username-Password-Authentication",
+            )
+        )
+        return str(user["user_id"])
+    except Auth0Error as error:
+        # If user already exists in Auth0, no problem!
+        if error.status_code == 409:
+            users = auth0.users_by_email.search_users_by_email(email.lower())
+            return str(users[0]["user_id"])
```

Comment:
What if the user is not in our DB? Unlikely given the sequence of calls, but probably something to check.

Reply:
read the first line wrong. Ignore.